src/dune_engine/build_system.ml

```diff
@@ -2349,22 +2354,34 @@ let prefix_rules (prefix : unit Action_builder.t) ~f =
 
 module Alias = Alias0
 
-let process_exn_and_reraise exn =
-  let open Fiber.O in
-  let exn =
-    Exn_with_backtrace.map exn ~f:(fun exn ->
-        match exn with
+let process_exn exn =
```

Comment:
I thought it was weird that we have an extra error processing layer here, since there are already two good places we could do it:

- in memo, as the error is raised: https://github.com/aalekseyev/dune/pull/2 (memo expect tests need fixing because the cycle error changed)
- in Report_error, as the error is reported: https://github.com/aalekseyev/dune/pull/1

I did the "in memo" version first, but I think I like the Report_error version better.
Do you agree? Do you agree they are both better than the current situation?

Reply:
I like the Report_error version too, I think it's the best of all three.